### ISSUE TITLE
Can you add the patch for OpenFOAM-v1812?

### ISSUE THREAD

**Issue by friedenhe** [mrklein/openfoam-os-x#73]
We wonder if you can make a patch for OpenFOAM-v1812. Thanks!

**Comment by mrklein** [comment]
Yes, I can. But I do not see a reason why.

**Comment by friedenhe** [comment]
Great! We developed a discrete adjoint capability for design optimization with OpenFOAM (check [here](dafoam.github.io)). Our code is based on OpenFOAM-v1812 and it is hard for us to upgrade to a newer OF version. We tried to follow the wiki and managed to compile OF on MacOS. However, it seems that the oldest supported version is v1912, so we wonder if you can add the v1812 patch. Thanks again!

**Comment by mrklein** [comment]
Well, I see the reason why you need it, I still do not see the reason why should I create it.

**Comment by olesenm** [comment]
> Great! We developed a discrete adjoint capability for design optimization with OpenFOAM (check [here](https://dafoam.github.io)). Our code is based on OpenFOAM-v1812 and it is hard for us to upgrade to a newer OF version. We tried to follow the wiki and managed to compile OF on MacOS. However, it seems that the oldest supported version is v1912, so we wonder if you can add the v1812 patch. Thanks again!

@friedenhe - instead of requesting patches for older versions of OpenFOAM, you would do better to see what is actually needed to update your code to a newer version of OpenFOAM. In addition to the [upgrade guides](https://develop.openfoam.com/Development/openfoam/-/wikis/upgrade/upgrade), you can probably also refer to corresponding parts of the [OpenFOAM source code](https://develop.openfoam.com/Development/openfoam) to see which changes are needed.

If we compare the [upstream checkTools.C](https://develop.openfoam.com/Development/openfoam/-/blob/master/applications/utilities/mesh/manipulation/checkMesh/checkTools.C) and [your checkTools.C](https://github.com/mdolab/dafoam/blob/master/src/adjoint/DACheckMesh/checkTools.C), there is an obvious amount of overlap. You should easily be able to apply an update there, but preferably ***not*** remove the copyright and license information this time!!

**Comment by friedenhe** [comment]
@olesenm, thanks very much for the suggestions! We will add the copyright and license info back. Yes, we are trying to upgrade our code with higher versions of OpenFOAM. The main DAFoam code should be straightforward to upgrade, the tricky part is that we used an automatic differentiation tool to overload all data types in OpenFOAM and differentiate its source code on this [OpenFOAM-v1812-AD](https://github.com/DAFoam/OpenFOAM-v1812-AD) repo. Upgrading that repo for a higher version of OpenFOAM may need some work. You left a [message](https://github.com/DAFoam/OpenFOAM-v1812-AD/issues/7) for us and suggest we rebase our code to upstream. We've just started doing that and may need your help. I left a message for you there. Thanks!
